### ISSUE TITLE
Bracket matching does not work as expected if the opening brace symbol is contained in the closing brace symbol (e.g. sequence endsequence)

### ISSUE THREAD

**Issue by eventhelix** [microsoft/vscode#11856]
- VSCode Version: 1.5.1
- OS Version: Windows 10

Steps to Reproduce:
1. Add custom bracket definitions as shown

``` js
{
    "comments": {
        // symbol used for single line comment. Remove this entry if your language does not support line comments
        "lineComment": "//",
        // symbols used for start and end a block comment. Remove this entry if your language does not support block comments
        "blockComment": [ "/*", "*/" ]
    },
    // symbols used as brackets
    "brackets": [
        ["(*", "*)"],       
        ["[*", "*]"],
        ["{", "}"],
        ["[", "]"],
        ["(", ")"],
        ["sequence", "endsequence"],
        ["feature", "endfeature"]   
    ],
    // symbols that are auto closed when typing
    "autoClosingPairs": [
        ["(*", "*)"],       
        ["[*", "*]"],
        ["{", "}"],
        ["\"", "\""],
        ["'", "'"],
        ["sequence", "endsequence"],
        ["feature", "endfeature"]  
    ],
    // symbols that that can be used to surround a selection
    "surroundingPairs": [
        ["(*", "*)"],       
        ["[*", "*]"],
        ["{", "}"],
        ["[", "]"],
        ["(", ")"],
        ["\"", "\""],
        ["'", "'"],
        ["sequence", "endsequence"],
        ["feature", "endfeature"]  
    ]
}
```
1.  Brace matching (Control+Shift+) for nested sequence blocks does not work as expected. For the following text: 

```
sequence "outer"
     sequence "inner"
     endsequence
endsequence
```

A brace match command on the outer endsequence, mistakenly selects the sequence text in the inner endsequence.

**Comment by eventhelix** [comment]
Another scenario to consider 
```json
{
    "autoClosingPairs": [
        ["(*", "*)"],       
        ["[*", "*]"],
        ["{", "}"],
        ["[", "]"],
        ["(", ")"],
        ["\"", "\""],
        ["'", "'"],
        ["sequence", "endsequence"],
        ["feature", "endfeature"]  
    ]
}
```

Here brace matching currently does not work for (* *) and [* *] as () and [] are also defined. 

The behavior could be: If you type a (, the editor matches with a ). If you add *, the editor brace matches to (**)

**Comment by alexandrudima** [comment]
Most of the cases are fixed with 91bd149, but not 100% all of them, like intermingling bracket types. e.g. 
```
[* [ asd *] ]
```
